sdk/storage/Azure.Storage.Common/tests/Shared/TestExtensions.cs

```diff
@@ -65,6 +65,82 @@ public static Uri ToHttp(this Uri uri)
+        /// <remarks>Using HTTPS to connect to the storage services is recommended.</remarks>
+        internal static StorageConnectionString CreateStorageConnectionString(object storageCredentials, string endpointSuffix, bool useHttps) =>
+            CreateStorageConnectionString(storageCredentials, storageCredentials is StorageSharedKeyCredential sharedKeyCredentials ? sharedKeyCredentials.AccountName : default, endpointSuffix, useHttps);
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="StorageConnectionString"/> class using the specified
+        /// credentials and endpoint suffix, and specifies whether to use HTTP or HTTPS to connect to the storage services.
+        /// </summary>
+        /// <param name="storageCredentials">A StorageCredentials object.</param>
+        /// <param name="accountName">The name of the account.</param>
+        /// <param name="endpointSuffix">The DNS endpoint suffix for all storage services, e.g. "core.windows.net".</param>
+        /// <param name="useHttps"><c>true</c> to use HTTPS to connect to storage service endpoints; otherwise, <c>false</c>.</param>
+        /// <param name="setEndpoint">Whether or not to set the endpoint properties on the ConnectionString.</param>
+        /// <remarks>Using HTTPS to connect to the storage services is recommended.</remarks>
+        internal static StorageConnectionString CreateStorageConnectionString(
+            object storageCredentials,
+            string accountName,
+            string endpointSuffix = Constants.ConnectionStrings.DefaultEndpointSuffix,
+            bool useHttps = true)
+        {
+            var conn = new StorageConnectionString(storageCredentials, default, default, default);
+            if (storageCredentials == null)
+            {
+                throw Errors.ArgumentNull(nameof(storageCredentials));
+            }
+
+            if (storageCredentials is StorageSharedKeyCredential sharedKeyCredentials && !string.IsNullOrEmpty(sharedKeyCredentials.AccountName))
+            {
+                if (string.IsNullOrEmpty(accountName))
+                {
+                    accountName = sharedKeyCredentials.AccountName;
+                }
+                else
+                {
+                    if (string.Compare(sharedKeyCredentials.AccountName, accountName, System.StringComparison.Ordinal) != 0)
+                    {
+                        throw Errors.AccountMismatch(sharedKeyCredentials.AccountName, accountName);
+                    }
+                }
+            }
+
+            if (accountName == default)
+            {
+                throw Errors.ArgumentNull(nameof(accountName));
+            }
+
+            conn._accountName = accountName;
+            var sasToken = (storageCredentials is SharedAccessSignatureCredentials sasCredentials) ? sasCredentials.SasToken : default;
+
+            var protocol = useHttps ? "https" : "http";
```

Comment:
Please consider using `Uri.UriSchemeHttps` and `Uri.UriSchemeHttp` instead of hard coding https and http.

Reply:
Actually, we have these defined already within Constants so I can use those.

Reply:
Can you look into changing the constants to use the Uri.UriSchemeHttp(s) properties?  This is a nit, but if it's already there in a const as part of .NET, then we should use it.

Reply:
I think this is getting a little outside the scope of this PR.

Reply:
I will hold off on this. One consideration is that we try to group all constants in one file for discoverability. I can't redirect to the Uri property either because our field is a "const".